odoo/orm/query.py

```diff
@@ -184,9 +216,10 @@ def select(self, *args: str | SQL) -> SQL:
             SQL(" ORDER BY %s", self._order) if self._order else _SQL_EMPTY,
             SQL(f" LIMIT {int(self.limit)}") if self.limit is not None else _SQL_EMPTY,
             SQL(f" OFFSET {int(self.offset)}") if self.offset else _SQL_EMPTY,
+            SQL(" %s", self.lock_parameters) if self.lock_parameters else _SQL_EMPTY,
         )
 
-    def subselect(self, *args: str | SQL) -> SQL:
+    def subselect(self, *args: str | SQL, query: Query | None = None) -> SQL:
```

Comment:
It seems like you are trying to move all CTE to the top query, which is not required in SQL (but may be needed for your optimization). I am thinking if we can have a simpler API has less side-effects and does not require updating other parts of the ORM (we will forget at some point adding that parameter as it is not obvious why it is here).